### ISSUE TITLE
Editing/Removing domainname doesn't work

### ISSUE THREAD

**Issue by fonfon** [freedombox/Plinth#301]
Entering a domainname (in sys - 'Configure') works, but I then cannot remove or edit it anymore.
I do this by submitting the form with the changed or empty input field 'Domain Name'.

It seems like socket.getfqdn() does not read `/etc/hosts` but tries to resolve via DNS which doesn't work in my case with VM and a domain that doesn't exist. (see also https://bugs.python.org/issue5004).
So when changing the domain name there is often a mismatch between the domain in `/etc/hosts` and the domain displayed in Plinth. (And the future name services module which contains accurate information.)

Content of /etc/hosts after inserting a hostname that does not exist:

```
127.0.0.1       localhost
127.0.1.1       freedombox.box.doesnotexist.at freedombox
```

Content of /etc/hosts after trying to remove the hostname:

```
127.0.0.1       localhost
127.0.1.1       freedombox. freedombox
```

At this stage, Plinth still displays `box.doesnotexist.at`. Also, if I change it via the form to `box.nodomain.at`, Plinth updates `/etc/hosts` but still displays `box.doesnotexist.at`.

**Comment by fonfon** [comment]
This is a duplicate of #262. I'll leave it open for now as contains additional information.

**Comment by jvalleroy** [comment]
Editing the domain name now works. But we still have an issue when removing, it leaves "freedombox." in /etc/hosts.

**Comment by mridulnagpal** [comment]
I would like to work on this issue

**Comment by mridulnagpal** [comment]
@jvalleroy I did not completely get the issue what is the problem exacty?

**Comment by jvalleroy** [comment]
@mridulnagpal Here is a test procedure that shows the issue.
1) Starting with a fresh FreedomBox vagrant image:
Hostname: freedombox
Domain Name: (empty)
```
$ cat /etc/hosts
127.0.0.1	localhost
127.0.1.1	freedombox
::1		localhost ip6-localhost ip6-loopback
ff02::1		ip6-allnodes
ff02::2		ip6-allrouters
```
2) Set domain name to "example.com":
```
$ cat /etc/hosts
127.0.0.1	localhost
127.0.1.1       freedombox.example.com freedombox
::1		localhost ip6-localhost ip6-loopback
ff02::1		ip6-allnodes
ff02::2		ip6-allrouters
```
3) Set domain name to empty "":
```
$ cat /etc/hosts
127.0.0.1	localhost
127.0.1.1       freedombox. freedombox
::1		localhost ip6-localhost ip6-loopback
ff02::1		ip6-allnodes
ff02::2		ip6-allrouters
```
When a domain name is set and then cleared, /etc/hosts should be set back to match the original from step 1. So in the 2nd line, we should remove the "freedombox.".

**Comment by mridulnagpal** [comment]
@jvalleroy https://github.com/freedombox/Plinth/pull/642
I have resolved the issue now it will remove $hostname. from the second line and only print the hostname

**Comment by mridulnagpal** [comment]
@jvalleroy @fonfon You may now close the issue as #642 has been merged and the issue has been resolved

**Comment by jvalleroy** [comment]
Fixed in #642.
